Userland/Applications/PixelPaint/Layer.cpp

```diff
@@ -198,7 +198,11 @@ void Layer::resize(Gfx::IntSize const& new_size, Gfx::IntPoint const& new_locati
         auto dst = Gfx::Bitmap::try_create(Gfx::BitmapFormat::BGRA8888, new_size).release_value_but_fixme_should_propagate_errors();
         Gfx::Painter painter(dst);
 
-        painter.draw_scaled_bitmap(dst_rect, *m_content_bitmap, src_rect, 1.0f, scaling_mode);
+        if (should_rescale) {
+            painter.draw_scaled_bitmap(dst_rect, *m_content_bitmap, src_rect, 1.0f, scaling_mode);
+        } else {
+            painter.draw_scaled_bitmap(src_rect, *m_content_bitmap, src_rect, 1.0f, scaling_mode);
```

Comment:
If we're not scaling the image, we can skip the scaling calculations by calling `painter.blit()` instead.